### ISSUE TITLE
[NvimTree] Unrecognized git state "DD" | Unrecognized git state "AU"

### ISSUE THREAD

**Issue by jens1205** [kyazdani42/nvim-tree.lua#496]
After having started a git rebase, which resulted in some conflicts, starting nvim shows one of the following errors (sometimes the first, sometimes the second) when trying to open one of the conflicted files:
- [NvimTree] Unrecognized git state "DD"
- [NvimTree] Unrecognized git state "AU"

Here is the current git status:
```
git rebase --continue                
internal/order-update-client/usecases/usecase_test.go: needs merge
internal/order-update-client/usecases/usecases_test.go: needs merge
internal/orderservice/usecases/usecase_test.go: needs merge
You must edit all merge conflicts and then
mark them as resolved using git add

git status           
interactive rebase in progress; onto 5753193
Last commands done (12 commands done):
   pick f703059 next
   pick 4df6b1e review comments jens - updating order-update schema file
  (see more in file .git/rebase-merge/done)
Next commands to do (14 remaining commands):
   pick dfa9817 updating order-change-event to v1.0.5
   pick 28aff82 moving Test_generateHorizonEvent to TestUseCases_FulfillmentItemsUpdateSuccess
  (use "git rebase --edit-todo" to view and edit)
You are currently rebasing branch 'feature/pom-1073' on '5753193'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   orderservice/controller/orderupdateclient/controller.go
	modified:   orderservice/controller/orderupdateclient/router.go
	modified:   orderservice/usecases/usecases.go
	modified:   ../json-model/order-change-event/v1/order-change-event

Unmerged paths:
  (use "git restore --staged <file>..." to unstage)
  (use "git add/rm <file>..." as appropriate to mark resolution)
	both deleted:    order-update-client/usecases/usecase_test.go
	added by them:   order-update-client/usecases/usecases_test.go
	added by us:     orderservice/usecases/usecase_test.go

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   ../json-model/order-change-event/v1/order-change-event (new commits)
```

The error message also kindly asks me to open an issue - so here it is :-)

**Comment by jens1205** [comment]
Ok, this error is strange. I realized that I have been in the wrong directory when I tried to open it in nvim - and the file was not existing.

So I tried to open - being already in directory internal - `nvim internal/order-update-client/usecases/usecases_test.go`.

So, being in a git repository directory which is in conflicted state, trying to open a non-exstant file with e.g. `nvim fajkldfja/fdjakldfjkl.txt` leads to the error.

The error cannot be reproduced in a git repository directory with a clean working tree.
